src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -191,3 +218,80 @@ impl<Src: OpendalSource> OpendalReader<Src> {
         }
     }
 }
+
+/// Since the `Reader` does not implement `tokio::io::AsyncRead` after `OpenDAL` 0.47, we construct a struct `ParquetFileReader` that implements `AsyncFileReader`, which is used as a parameter of `ParquetRecordBatchStreamBuilder`.
+/// The following code refers to <`https://github.com/icelake-io/icelake/blob/07d53893d7788b4e41fc11efad8a6be828405c31/icelake/src/io/scan.rs#L196`>, we can remove the following implementation after `ParquetFileReader` is changed to a public struct in icelake.
+pub struct ParquetFileReader {
+    op: Operator,
+    path: String,
+}
+
+impl AsyncFileReader for ParquetFileReader {
```

Comment:
I did notice the `into_futures_async_read ` interface, but it will return a `Reader` which `impl  futures::AsyncRead`, while we need an Reader that `impl  tokio::io::AsyncRead`, so just add another struct. 

This also applies to `Writer`: `into_futures_async_write` will return a `Writer` which `impl  futures::AsyncWrite`, not `impl  tokio::io::AsyncWrite`.

Reply:
You may try this: https://docs.rs/tokio-util/latest/tokio_util/compat/index.html